src/bulk/common.ts

```diff
@@ -1346,13 +1346,13 @@ function handleEarlyError(
   err?: AnyError,
   callback?: Callback<BulkWriteResult>
 ): Promise<BulkWriteResult> | void {
-  const Promise = PromiseProvider.get();
   if (typeof callback === 'function') {
     callback(err);
     return;
   }
 
-  return Promise.reject(err);
+  const PromiseConstructor = PromiseProvider.get() ?? Promise;
+  return PromiseConstructor.reject(err);
 }
```

Comment:
There are only two instances of handleErrorEarly in the entire repo, both in BulkOperationBase.execute().  How would you feel about just refactoring bulkOperationBase to use maybeCallback and getting rid of this helper entirely?

If we don't want to do that - why not just use maybeCallback?  This helper can be written like follows
```
  return maybeCallback(() => {
    throw err;
  }, callback);
```

Reply:
Sure refactor sounds good